opengrok-indexer/src/main/java/org/opengrok/indexer/configuration/Group.java

```diff
@@ -301,12 +304,12 @@ public static Group getByName(String name) {
      * the project's description.
      *
      * @param project the project
-     * @param groups set of groups
+     * @param map set of groups
      * @return set of groups matching the project
      */
-    public static Set<Group> matching(Project project, Set<Group> groups) {
-        Set<Group> copy = new TreeSet<>(groups);
-        copy.removeIf(g -> !g.match(project));
+    public static Map<String, Group> matching(Project project, Map<String, Group> map) {
```

Comment:
Do you need to change the return value to map? It seems that the only usage just calls `.values()` immediately.